src/Psalm/Internal/Analyzer/ProjectAnalyzer.php

```diff
@@ -1179,10 +1183,27 @@ public function refactorCodeAfterCompletion(array $to_refactor): void
      */
     public function setPhpVersion(string $version, string $source): void
     {
-        if (!preg_match('/^(5\.[456]|7\.[01234]|8\.[012])(\..*)?$/', $version)) {
-            throw new UnexpectedValueException('Expecting a version number in the format x.y');
+        if (!preg_match('/' . self::PHP_VERSION_REGEX . '/', $version)) {
+            fwrite(
+                STDERR,
+                'Expecting a version number in the format x.y or x.y.z'
+                . PHP_EOL,
+            );
+            exit(1);
         }
 
+        if (!preg_match('/' . self::PHP_SUPPORTED_VERSIONS_REGEX . '/', $version)) {
+            fwrite(
+                STDERR,
+                'Psalm supports PHP version ">=5.4". The specified version '
+                . $version
+                . " is either not supported or doesn't exist."
+                . PHP_EOL,
+            );
+            exit(1);
+        }
```

Comment:
`exit(1)` is reserved for when Psalm finds some issues with the codebase it inspects, see https://psalm.dev/docs/running_psalm/command_line_usage/#exit-status
 
Also, this method is used in a lot of tests, and exception is so much more useful in tests compared to outright crash. Please keep the exception, but catch it in `CliUtils::initPhpVersion()` and `exit(2)` there instead.

Reply:
Actually it's the other way around. `exit(1)` is for general problems, and `exit(2)` is for successful run that found some code issues.

Reply:
Sorry for the confusion. I'll fix it myself.